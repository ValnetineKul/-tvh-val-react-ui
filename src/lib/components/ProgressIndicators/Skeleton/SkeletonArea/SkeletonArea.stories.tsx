import { createMeta, createTemplate } from '../../../../story-utils';
import SkeletonArea from './SkeletonArea';

const Template = createTemplate(SkeletonArea);

export const Primitive = Template.bind({});
Primitive.args = {
  width: 200,
  height: 200,
};
Primitive.storyName = 'SkeletonArea';

export default createMeta({
  component: SkeletonArea,
  title: 'Components/ProgressIndicators/Skeleton/SkeletonArea',
});
