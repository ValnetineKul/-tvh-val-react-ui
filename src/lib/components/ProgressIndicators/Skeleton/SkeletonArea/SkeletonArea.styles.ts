import type { Surface } from '../../../../../themes/core';
import { makeStyles } from '../../../../../themes/core';
import { skeletonAnimationStyles } from '../helper';

type Props = {
  onSurface: Surface;
  width: string | number;
  height: string | number;
};

const useStyles = makeStyles<Props>({
  name: 'SkeletonArea',
})((theme, { onSurface, width, height }) => ({
  root: {
    width,
    height,
    ...skeletonAnimationStyles(
      onSurface === '200' ? theme.color['bg/surface300/default'] : theme.color['bg/surface200/default'],
      onSurface === '200' ? theme.color['bg/surface400/default'] : theme.color['bg/surface150/default']
    ),
  },
}));

export default useStyles;
