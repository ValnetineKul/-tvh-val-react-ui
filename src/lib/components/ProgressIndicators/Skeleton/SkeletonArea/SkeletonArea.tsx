import type { FC } from 'react';
import React from 'react';
import { Skeleton } from '@mui/material';
import { useSurface } from '../../../../../themes/core';
import useStyles from './SkeletonArea.styles';

export interface SkeletonAreaProps {
  width: string | number;
  height: string | number;
  className?: string;
}

const SkeletonArea: FC<SkeletonAreaProps> = ({ width = '100%', height, className }) => {
  const { color: onSurface } = useSurface();
  const { classes, cx } = useStyles({ onSurface, width, height });

  return (
    <Skeleton
      data-testid="skeleton-area"
      className={cx(classes.root, className)}
      variant="rectangular"
      animation="wave"
    />
  );
};

export default SkeletonArea;
