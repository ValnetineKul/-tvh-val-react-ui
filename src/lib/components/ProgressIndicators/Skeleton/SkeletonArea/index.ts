import SkeletonArea from './SkeletonArea';

export type { SkeletonAreaProps } from './SkeletonArea';
export default SkeletonArea;
