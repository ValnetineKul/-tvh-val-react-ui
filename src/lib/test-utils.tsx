import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
