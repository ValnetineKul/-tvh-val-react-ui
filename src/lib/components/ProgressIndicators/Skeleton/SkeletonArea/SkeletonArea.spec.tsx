import React from 'react';
import SkeletonArea from './SkeletonArea';
import { render } from '../../../../test-utils';

describe('SkeletonArea', () => {
  it('Should render correctly', () => {
    const { container } = render(<SkeletonArea width="200px" height="200px" />);
    expect(container.firstChild).toMatchSnapshot('SkeletonArea');
  });
});
