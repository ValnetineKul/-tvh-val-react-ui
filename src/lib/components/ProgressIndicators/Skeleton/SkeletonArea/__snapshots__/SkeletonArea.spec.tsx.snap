// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonArea Should render correctly: SkeletonArea 1`] = `
<span
  class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave tss-1vqhqgo-SkeletonArea-root mui-1slz293-MuiSkeleton-root"
  data-testid="skeleton-area"
/>
`;
